nemo/collections/asr/parts/k2/graph_compilers.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copyright (c) 2020, Xiaomi CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple
+
+import k2
+import torch
+
+from nemo.collections.asr.parts.k2.topologies import build_topo
+from nemo.collections.asr.parts.k2.utils import compose_with_self_loops, intersect_with_self_loops
+
+
+class CtcTrainingTopologyCompiler(object):
+    """Default training graph compiler.
+    It applies its topology to the input token sequence to compile the numerator graph.
+    
+    Based on https://github.com/k2-fsa/snowfall/blob/master/snowfall/training/ctc_graph.py
+    """
+
+    def __init__(
+        self,
+        num_classes: int,
+        topo_type: str = "default",
+        topo_with_selfloops: bool = True,
+        device: torch.device = torch.device("cpu"),
+    ):
+        self.topo_type = topo_type
```

Comment:
Import guard check here.

Reply:
Done in [8d3e482](https://github.com/NVIDIA/NeMo/pull/3351/commits/8d3e48215629824b59a020526a0444c944b2d1c2)